### PR TITLE
win support

### PR DESCRIPTION
**Pull request by haiyangwu**
using the win style socket path

### CODE REVIEW

**Comment by ibc** [comment]
Amazing. Let's go first for the PR in mediasoup and come back to this one later.

**Comment by ibc** [comment]
This is absolutely amazing, thanks!

However we have zero experience in Windows so we'll make lot of questions and may request some changes. Let's go step by step.

File: server/lib/interactiveServer.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ const fs = require('fs');
 const mediasoup = require('mediasoup');
 const colors = require('colors/safe');
 const pidusage = require('pidusage');
+const os = require('os');
+const path = require('path');
```
Comment (ibc) [inline server/lib/interactiveServer.js]: Let's move system modules to the top of the file.

File: server/lib/interactiveServer.js [hunk 2]
```diff
@@ -15,7 +17,9 @@ const consumers = new Map();
 const dataProducers = new Map();
 const dataConsumers = new Map();
 
-const SOCKET_PATH = '/tmp/mediasoup-demo.sock';
+const SOCKET_PATH_UIX = '/tmp/mediasoup-demo.sock';
+const SOCKET_PATH_WIN = path.join('\\\\?\\pipe', process.cwd(), 'mediasoup-demo');
+const SOCKET_PATH = os.platform() === 'win32'? SOCKET_PATH_WIN : SOCKET_PATH_UIX
```
Comment (ibc) [inline server/lib/interactiveServer.js]: Let's move these constants after `require()` lines above.

**Review by ibc — CHANGES_REQUESTED** [review]

**Review by TemaSM — APPROVED** [review]
LGTM ✔️

File: .gitignore [hunk 3]
```diff
@@ -5,3 +5,5 @@ node_modules/
 /server/public/
 /server/certs/
 !/server/certs/mediasoup-demo.localhost.*
+
+/.vscode/
```
Comment (ibc) [inline .gitignore]: Add line jump at the end of the file.

File: server/lib/interactiveClient.js [hunk 4]
```diff
@@ -1,6 +1,10 @@
 const net = require('net');
+const os = require('os');
+const path = require('path');
 
-const SOCKET_PATH = '/tmp/mediasoup-demo.sock';
+const SOCKET_PATH_UIX = '/tmp/mediasoup-demo.sock';
```
Comment (ibc) [inline server/lib/interactiveClient.js]: Please rename (fix it) to `SOCKET_PATH_UNIX`.

File: server/lib/interactiveServer.js [hunk 5]
```diff
@@ -6,6 +8,10 @@ const mediasoup = require('mediasoup');
 const colors = require('colors/safe');
 const pidusage = require('pidusage');
 
+const SOCKET_PATH_UIX = '/tmp/mediasoup-demo.sock';
```
Comment (ibc) [inline server/lib/interactiveServer.js]: Please rename (fix it) to SOCKET_PATH_UNIX.

**Review by ibc — CHANGES_REQUESTED** [review]

**Review by ibc — APPROVED** [review]

**Comment by ibc** [comment]
Thanks a lot, merged.

**Pull request merged**